python/pyspark/sql/avro/functions.py

```diff
@@ -26,7 +26,7 @@
 
 
 @since(3.0)
-def from_avro(data, jsonFormatSchema, options={}):
+def from_avro(data, jsonFormatSchema, options=None):
```

Comment:
We should change annotations from

```python
def from_avro(
    data: ColumnOrName, jsonFormatSchema: str, options: Dict[str, str] = ...
) -> Column: ..
```

to

```python
def from_avro(
    data: ColumnOrName, jsonFormatSchema: str, options: Optional[Dict[str, str]] = ...
) -> Column: ..
```

Reply:
Thanks. Just curious, how can we check this? Mypy has been added, and I would expect that this would be caught.

Reply:
>  Mypy has been added, and I would expect that this would be caught.

As far as I recall this behavior depends on the MyPy flags. In particular see [Disabling strict optional checking](https://mypy.readthedocs.io/en/stable/kinds_of_types.html#no-strict-optional).

So adding `no_implicit_optional` to `mypy.ini` should catch this.

Reply:
FYI I am trying to establish optimal set of flags for mypy config, but it is a bit slowish work ‒ unlike standalone stubs, we have to correct for internal modules, tests and such, and I'd prefer to avoid atomic options (ignoring all errors there).

Reply:
Thanks! 👍  Yes, definitely a process that we have to go through. Let's break this up into smaller pieces.